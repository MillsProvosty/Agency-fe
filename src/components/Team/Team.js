import React from "react";
import "./Team.scss";
import { FaLinkedin, FaGithub } from "react-icons/fa";
import Nav from "../Nav/Nav";
import styled from "styled-components";

const Program = styled.p`
  margin: ${props => props.name ? '50px' : '0px'};
  font-weight: ${props => props.name ? 'bold': ''};
  font-size: ${props => props.name ? '2em' : '1.5em'}
`;

const ProgramSection = styled.section`
  display: flex;
  margin-bottom: 50px;
`;

const Member = styled.section`
  width: 50%;
`;

const URLSection = styled.section`
  display: flex;
  width: 70%;
  justify-content: space-around;
  margin: 0px auto;
`;

const Image = styled.img`
  width: 15em;
  border-radius: 50%;
  border: 2px solid darkblue;
`;


const Team = () => {
  return (
    <section className="Team">
      <Nav />
      <Program name className="program">Back End:</Program>
      <ProgramSection className="back">
        <Member className="mills">
          <Program>Mills Provosty</Program>
          <Image
            src="https://avatars0.githubusercontent.com/u/45856797?s=400&v=4"
            alt="developer"
          />
          <URLSection className="urls">
            <a className="url" href="https://github.com/MillsProvosty">
              <FaGithub size={32} />
            </a>
            <a
              className="url"
              href="https://www.linkedin.com/in/mills-provosty/"
            >
              <FaLinkedin size={32} />
            </a>
          </URLSection>
        </Member>
        <Member className="tay">
          <Program>Tay De Herrera</Program>
          <Image
            src="https://avatars2.githubusercontent.com/u/47490116?s=400&v=4"
            alt="developer"
          />
          <URLSection className="urls">
            <a className="url" href="https://github.com/tayjames">
              <FaGithub size={32} />
            </a>
            <a
              className="url"
              href="https://www.linkedin.com/in/tay-deherrera/"
            >
              <FaLinkedin size={32} />
            </a>
          </URLSection>
        </Member>
      </ProgramSection>
      <Program name className="program">Front End:</Program>
      <ProgramSection className="front">
        <Member className="aidan">
          <Program>Aidan McKay</Program>
          <Image
            src="https://avatars1.githubusercontent.com/u/27786530?s=400&v=4"
            alt="developer"
          />
          <URLSection className="urls">
            <a className="url" href="https://github.com/JellyBeans1312">
              <FaGithub size={32} />
            </a>
            <a className="url" href="https://www.linkedin.com/in/aidan-mckay/">
              <FaLinkedin size={32} />
            </a>
          </URLSection>
        </Member>
        <Member className="greg">
          <Program>Gregory Anderson</Program>
          <Image
            src="https://avatars1.githubusercontent.com/u/30326085?s=400&v=4"
            alt="developer"
          />
          <URLSection className="urls">
            <a className="url" href="https://github.com/gregoryanderson">
              <FaGithub size={32} />
            </a>
            <a
              className="url"
              href="https://www.linkedin.com/in/gregandersondev/"
            >
              <FaLinkedin size={32} />
            </a>
          </URLSection>
        </Member>
      </ProgramSection>
    </section>
  );
};

export default Team;
