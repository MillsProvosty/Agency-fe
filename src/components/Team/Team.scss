.Team {
  & .program {
    margin: 50px;
    font-weight: bold;
  }

  & .back,
  .front {
    display: flex;
    margin-bottom: 50px;
  }

  & .mills,
  .tay,
  .greg,
  .aidan {
    width: 50%;
  }

  & .urls {
    display: flex;
    width: 70%;
    justify-content: space-around;
    margin: 0px auto;
    & .url {
      font-size: 1.5em;
    }
  }

  & img {
      width: 15em;
      border-radius: 50%;
      border: 2px solid darkblue
  }
}
