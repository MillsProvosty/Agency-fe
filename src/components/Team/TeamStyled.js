import styled from "styled-components";

export const Program = styled.p`
  background-color: white;
  padding: ${props => props.name ? '50px' : '0px'};
  font-weight: ${props => props.name ? 'bold': ''};
  font-size: ${props => props.name ? '2em' : '1.5em'}
  margin: 0;
`;

export const ProgramSection = styled.section`
  background-color: white;
  display: flex;
  padding-bottom: 50px;
`;

export const Member = styled.section`
  width: 50%;
`;

export const URLSection = styled.section`
  display: flex;
  width: 70%;
  justify-content: space-around;
  margin: 0px auto;
`;

export const Image = styled.img`
  width: 15em;
  border-radius: 50%;
  border: 2px solid #37474e;
`;
