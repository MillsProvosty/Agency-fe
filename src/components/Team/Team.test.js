import React from 'react';
import { shallow } from 'enzyme';
import Team from './Team';

describe('Team', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Team/>
    )
    expect(wrapper).toMatchSnapshot()
  }) 
})