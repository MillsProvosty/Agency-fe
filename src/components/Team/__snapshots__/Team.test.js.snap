// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team should match snapshot 1`] = `
<section
  className="Team"
>
  <Nav />
  <styled.p
    className="program"
    name={true}
  >
    Back End:
  </styled.p>
  <styled.section
    className="back"
  >
    <styled.section
      className="mills"
    >
      <styled.p>
        Mills Provosty
      </styled.p>
      <styled.img
        alt="developer"
        src="https://avatars0.githubusercontent.com/u/45856797?s=400&v=4"
      />
      <styled.section
        className="urls"
      >
        <a
          className="url"
          href="https://github.com/MillsProvosty"
        >
          <FaGithub
            size={32}
          />
        </a>
        <a
          className="url"
          href="https://www.linkedin.com/in/mills-provosty/"
        >
          <FaLinkedin
            size={32}
          />
        </a>
      </styled.section>
    </styled.section>
    <styled.section
      className="tay"
    >
      <styled.p>
        Tay DeHerrera
      </styled.p>
      <styled.img
        alt="developer"
        src="https://avatars2.githubusercontent.com/u/47490116?s=400&v=4"
      />
      <styled.section
        className="urls"
      >
        <a
          className="url"
          href="https://github.com/tayjames"
        >
          <FaGithub
            size={32}
          />
        </a>
        <a
          className="url"
          href="https://www.linkedin.com/in/tay-deherrera/"
        >
          <FaLinkedin
            size={32}
          />
        </a>
      </styled.section>
    </styled.section>
  </styled.section>
  <styled.p
    className="program"
    name={true}
  >
    Front End:
  </styled.p>
  <styled.section
    className="front"
  >
    <styled.section
      className="aidan"
    >
      <styled.p>
        Aidan McKay
      </styled.p>
      <styled.img
        alt="developer"
        src="https://avatars1.githubusercontent.com/u/27786530?s=400&v=4"
      />
      <styled.section
        className="urls"
      >
        <a
          className="url"
          href="https://github.com/JellyBeans1312"
        >
          <FaGithub
            size={32}
          />
        </a>
        <a
          className="url"
          href="https://www.linkedin.com/in/aidan-mckay/"
        >
          <FaLinkedin
            size={32}
          />
        </a>
      </styled.section>
    </styled.section>
    <styled.section
      className="greg"
    >
      <styled.p>
        Gregory Anderson
      </styled.p>
      <styled.img
        alt="developer"
        src="https://avatars1.githubusercontent.com/u/30326085?s=400&v=4"
      />
      <styled.section
        className="urls"
      >
        <a
          className="url"
          href="https://github.com/gregoryanderson"
        >
          <FaGithub
            size={32}
          />
        </a>
        <a
          className="url"
          href="https://www.linkedin.com/in/gregandersondev/"
        >
          <FaLinkedin
            size={32}
          />
        </a>
      </styled.section>
    </styled.section>
  </styled.section>
</section>
`;
